index.js

```diff
@@ -0,0 +1,330 @@
+//declaration of color events
+function illumine(event) {
+    event.target.style.backgroundColor = '#a1f4f3';
+    event.target.style.border = '3px solid #42aff7';
+    // event.target.addEventListener('mouseout', green)
+    }
+
+    function green(event) {
+        event.target.style.backgroundColor = '#d3ffd3';
+        event.target.style.border = '#5dff5d'
+        // event.target.addEventListener('mouseover', purple);
+    }
+
+    function yellow(event){
+        event.target.style.backgroundColor = '#ffff62';
+        event.target.style.border = '3px solid #9d9d00';
+        // event.target.addEventListener('mouseover', darken);
+    }
+
+    function purple(event){
+        event.target.style.backgroundColor = '#bb00bb';
+        event.target.style.border = '3px solid #ff6cff'
+        // event.target.addEventListener('mouseout', yellow)
+    }
+
+function darken(event) {
+    event.target.style.backgroundColor = 'white';
+    event.target.style.border = '3px solid black';
+    // event.target.addEventListener('mouseout', illumine);
+}
+
+// function to add to the unclicked cell 
+function clicked(event) {
+    event.target.textContent = currentPlayer;
+    // currentPlayer === 'x' ? 'o' : 'x';
+    event.target.removeEventListener('click', clicked);
+    event.target.addEventListener('click', alreadyClicked);
+    usedCellArray.push(event.target);
+    checkWin();
+
+    let win = checkWin();
+
+    if (win) {
+        if (currentPlayer === 'x') {
+            status.textContent = `${capitalize(playerOne)} wins!`
+        } else {
+            status.textContent = `${capitalize(playerTwo)} wins!`
+        }
+        stopPlay(cellArray)
+        clearInterval(interval);
+        clockCount = 0;
+        startBtn.disabled = true;
+        onePlayer.disabled = false;
+        twoPlayer.disabled = false;
+        playerOneName.disabled = false;
+        playerTwoName.disabled = false;
+    } else if (parseInt(usedCellArray.length) === 9) {
+        status.textContent = `It's a draw!`;
+        onePlayer.disabled = false;
+        twoPlayer.disabled = false;
+        playerOneName.disabled = false;
+        playerTwoName.disabled = false;
+        stopPlay(cellArray)
+        clearInterval(interval);
+        clockCount = 0;
+    } else {
+        if (gameMode === 'onePlayer') {
+            if (currentPlayer === 'x') {
+                stopPlay(cellArray);
+                currentPlayer = 'o';
+                status.textContent = `${capitalize(playerTwo)}'s turn!`
+                let randomCell = cellArray[Math.floor(Math.random() * 9)];
+                while (randomCell.textContent !== '' && randomCell) {
+                    randomCell = cellArray[Math.floor(Math.random() * 9)];
+                }
+                randomCell.removeEventListener('click', alreadyClicked)
+                randomCell.addEventListener('click', clicked);
+                setTimeout(() => {
+                    randomCell.click()
+                }, 1000);
```

Comment:
Lowering or removing this timer can fix your bug with the user being able to play on the computers turn.

Reply:
I thought I had fixed this issue by adding a function that removed all event listeners from the cells when it switches to player O's turn, then adding one back to the "randomCell" just before the computer click event, then adding the listeners back to the cells. I definitely remember explaining this function to Erin while writing it when we worked on this Sunday, and it being a fix for the bug. The only drawback was, if the player clicked on the cell the computer had chosen as the random cell, it would play the O there before the computer had time to click, and the computer would play the X in turn, but there was only a 1/9 chance of that happening, much slimmer than 100%.